src/daft-core/src/array/ops/utf8.rs

```diff
@@ -348,6 +350,19 @@ pub enum PadPlacement {
     Right,
 }
 
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct Utf8NormalizeOptions {
+    pub remove_punct: bool,
+    pub lowercase: bool,
+    pub nfd_unicode: bool,
+    pub white_space: bool,
+}
+
+const PUNCTUATION: [char; 32] = [
+    '!', '\"', '#', '$', '%', '&', '\'', '(', ')', '*', '+', ',', '-', '.', '/', ':', ';', '<',
+    '=', '>', '?', '@', '[', '\\', ']', '^', '_', '`', '{', '|', '}', '~',
+];
```

Comment:
Instead of hard-coding it, it may be cleaner to use Rust's [char::is_ascii_punctuation](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_punctuation)